Tests/Algorithm/AlgorithmRegisterIndicatorTests.cs

```diff
@@ -0,0 +1,153 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using NUnit.Framework;
+using Python.Runtime;
+using QuantConnect.Algorithm;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Indicators;
+using QuantConnect.Tests.Indicators;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Tests.Algorithm
+{
+    [TestFixture]
+    public class AlgorithmRegisterIndicatorTests
+    {
+        private Symbol _spy;
+        private QCAlgorithm _algorithm;
+        private IEnumerable<IGrouping<Type, Type>> _indicatorTestsGroups;
+
+        [SetUp]
+        public void Setup()
+        {
+            _algorithm = new QCAlgorithm();
+            _spy = _algorithm.AddEquity("SPY").Symbol;
+
+            _indicatorTestsGroups =
+                from type in GetType().Assembly.GetTypes()
+                where type.IsPublic && !type.IsAbstract
+                where
+                   typeof(CommonIndicatorTests<TradeBar>).IsAssignableFrom(type) ||
+                   typeof(CommonIndicatorTests<IBaseDataBar>).IsAssignableFrom(type) ||
+                   typeof(CommonIndicatorTests<IndicatorDataPoint>).IsAssignableFrom(type)
+                group type by type.BaseType.GetGenericArguments().FirstOrDefault();
+        }
+
+        [Test]
+        public void RegistersIndicatorProperly()
+        {
+            var expected = 0;
+            foreach (var group in _indicatorTestsGroups)
+            {
+                var key = group.Key;
+                foreach (var type in group)
+                {
+                    var indicatorTest = Activator.CreateInstance(type);
+                    if (key == typeof(IndicatorDataPoint))
+                    {
+                        var indicator = (indicatorTest as CommonIndicatorTests<IndicatorDataPoint>).Indicator;
+                        Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute, Field.Close));
+                    }
+                    else if (key == typeof(IBaseDataBar))
+                    {
+                        var indicator = (indicatorTest as CommonIndicatorTests<IBaseDataBar>).Indicator;
+                        Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute));
+                    }
+                    else if (key == typeof(TradeBar))
+                    {
+                        var indicator = (indicatorTest as CommonIndicatorTests<TradeBar>).Indicator;
+                        Assert.DoesNotThrow(() => _algorithm.RegisterIndicator(_spy, indicator, Resolution.Minute));
+                    }
+                    expected++;
+                    var actual = _algorithm.SubscriptionManager.Subscriptions.FirstOrDefault().Consolidators.Count;
+                    Assert.AreEqual(expected, actual);
```

Comment:
Perhaps the `expected++` belongs inside each of the type checks. We want to confirm each case has been hit. Similar comment for the below test as well.